examples/examples_test.go

```diff
@@ -0,0 +1,261 @@
+	// Remove braces from args when logging so users can see the complete call
+	logf("Executing %s %v", c, strings.Trim(fmt.Sprint(args), "[]"))
+
+	cmd := exec.Command(c, args...)
+	cmd.Env = os.Environ()
+
+	if input != "" {
+		cmd.Stdin = strings.NewReader(input)
+	}
+
+	var stderr, stdout bytes.Buffer
+	cmd.Stderr = &stderr
+	cmd.Stdout = &stdout
+	if err := cmd.Run(); err != nil {
+		logf("couldn't run command %s %v: %v, %s", c, args, err, stderr.String())
+		return "", err
+	}
+
+	return stdout.String(), nil
+}
+
+// eraseClusterTasks will erase all cluster tasks which do not get cleaned up by namespace.
+func eraseClusterTasks(logf formatLogger) {
+	if _, err := cmd(logf, "kubectl", []string{"delete", "--ignore-not-found=true", "clustertasks.tekton.dev", "--all"}, ""); err != nil {
+		logf("couldn't delete cluster tasks: %v", err)
+	}
+}
+
+// getYamls will look in the directory in examples indicated by version and run for yaml files
+func getYamls(t *testing.T, version, run string) []string {
+	t.Helper()
+	_, filename, _, _ := runtime.Caller(0)
+
+	// Don't read recursively; the only dir within these dirs is no-ci which doesn't
+	// want any tests run against it
+	files, err := ioutil.ReadDir(path.Join(path.Dir(filename), version, run))
+	if err != nil {
+		t.Fatalf("Couldn't read yaml files from %s/%s/%s: %v", path.Dir(filename), version, run, err)
+	}
+	yamls := []string{}
+	for _, f := range files {
+		if matches, _ := filepath.Match("*.yaml", f.Name()); matches {
+			yamls = append(yamls, f.Name())
+		}
+	}
+	return yamls
+}
+
+// replaceDockerRepo will look in the content f and replace the hard coded docker
+// repo with the on provided via the KO_DOCKER_REPO environment variable
+func replaceDockerRepo(t *testing.T, f string) string {
+	t.Helper()
+	r := os.Getenv("KO_DOCKER_REPO")
+	if r == "" {
+		t.Fatalf("KO_DOCKER_REPO must be set")
+	}
+	read, err := ioutil.ReadFile(f)
+	if err != nil {
+		t.Fatalf("couldnt read contents of %s: %v", f, err)
+	}
+	return strings.Replace(string(read), horribleHardCodedRegistry, r, -1)
+}
+
+// logRun will retrieve the entire yaml of run in namespace n and log it
+func logRun(t *testing.T, n, run string) {
+	t.Helper()
+	yaml, err := cmd(t.Logf, "kubectl", []string{"--namespace", n, "get", run, "-o", "yaml"}, "")
+	if err == nil {
+		t.Logf(yaml)
+	}
+}
+
+// pollRun will use kubectl to query the specified run in namespace n
+// to see if it has completed. It will timeout after timeoutSeconds.
+func pollRun(t *testing.T, n, run string, wg *sync.WaitGroup) {
+	t.Helper()
+	// instead of polling it might be faster to explore using --watch and parsing
+	// the output as it returns
+	for i := 0; i < (timeoutSeconds / sleepBetween); i++ {
+		status, err := cmd(t.Logf, "kubectl", []string{"--namespace", n, "get", run, "--output=jsonpath={.status.conditions[*].status}"}, "")
+		if err != nil {
+			t.Fatalf("couldnt get status of %s: %v", run, err)
```

Comment:
`t.Fatalf` does `t.FailNow` which calls `runtime.Goexit`, so `wg.Done()`  seems unecessary :upside_down_face:

Reply:
This means it will quit the test on this error… If that's not what we want, we need to use `t.Errorf`.